src/main/java/de/muenchen/isi/domain/service/stammdaten/FoerdermixStammService.java

```diff
@@ -53,11 +54,17 @@ public FoerdermixStammModel getFoerdermixStammById(final UUID id) throws EntityN
      *
      * @param foerdermix zum Speichern.
      * @return das gespeicherte {@link FoerdermixStammModel}.
+     * @throws UniqueViolationException falls die Bezeichnung {@link FoerdermixStammModel#getBezeichnung()} des Fördermixes bereits im gleichen Jahr {@link FoerdermixStammModel#getBezeichnungJahr()} vorhanden ist.
      */
-    public FoerdermixStammModel saveFoerdermixStamm(final FoerdermixStammModel foerdermix) {
+    public FoerdermixStammModel saveFoerdermixStamm(final FoerdermixStammModel foerdermix) throws UniqueViolationException {
         var entity = this.stammdatenDomainMapper.model2Entity(foerdermix);
-        entity = this.foerdermixStammRepository.save(entity);
-        return this.stammdatenDomainMapper.entity2Model(entity);
+        final var saved = this.foerdermixStammRepository.findByBezeichnungJahrIgnoreCaseAndBezeichnungIgnoreCase(foerdermix.getBezeichnungJahr(), foerdermix.getBezeichnung());
+        if (saved.isPresent()) {
+            throw new UniqueViolationException("Die Bezeichnung exisitiert bereits unter dem angegebenen Jahr. Bitte wählen Sie daher eine andere Bezeichnung und speichern Sie den Fördermix erneut.");
+        } else {
+            entity = this.foerdermixStammRepository.save(entity);
+            return this.stammdatenDomainMapper.entity2Model(entity);
```

Comment:
Bitte wie in ISI-446 (#20) anpassen, da es sonst wieder zum Bug führt und man nichts updaten oder speichern kann

Reply:
Durch Definition einer Unique-Constraint wird die Prüflogik in der Methode hinfällig.

https://github.com/it-at-m/isi-backend/pull/22#discussion_r1106931372

Reply:
angepasst